graph_weather/models/aurora/encoder.py

```diff
@@ -6,35 +6,41 @@
 """
 
 import torch.nn as nn
+from einops import rearrange, reduce, repeat
+from einops.layers.torch import Rearrange
 
 
 class Swin3DEncoder(nn.Module):
     def __init__(self, in_channels=1, embed_dim=96):
-        """
-        Initialize the Swin3DEncoder.
-
-        Args:
-            in_channels (int): Number of input channels (e.g., weather variable channels).
-            embed_dim (int): Embedding dimension for the transformer.
-        """
         super().__init__()
         self.conv1 = nn.Conv3d(in_channels, embed_dim, kernel_size=3, padding=1, stride=1)
         self.norm = nn.LayerNorm(embed_dim)
-        self.swin_transformer = nn.Transformer(embed_dim, num_encoder_layers=4)
-
+        self.swin_transformer = nn.Transformer(
+            d_model=embed_dim,
+            nhead=8,  # Standard number of heads
+            num_encoder_layers=4,
+            num_decoder_layers=4,
+            dim_feedforward=embed_dim * 4
+        )
+        self.embed_dim = embed_dim
     def forward(self, x):
-        """
-        Forward pass for the encoder.
-
-        Args:
-            x (torch.Tensor): Input tensor of shape (batch_size, channels, depth, height, width).
-
-        Returns:
-            torch.Tensor: Encoded features.
-        """
-        x = self.convolution(x)  # Convolution layer
-        x = self.normalization_layer(x)  # Normalization layer
-        x = self.transformer_encoder(x)  # Transformer encoder
+        # Apply 3D convolution: (batch, channels, depth, height, width) -> (batch, embed_dim, depth, height, width)
+        x = self.conv1(x)
+        
+        # Reshape for layer norm: (batch, embed_dim, depth, height, width) -> (batch, depth*height*width, embed_dim)
+        batch_size = x.shape[0]
+        x = x.permute(0, 2, 3, 4, 1).contiguous()  # -> (batch, depth, height, width, embed_dim)
```

Comment:
I would suggest going through all these kinds of `permute` `view` etc. and using einops for them, to be more consistent and more readable.